atlas/packages/app/src/shared/components/Button/Button.tsx

```diff
@@ -1,31 +1,46 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { SerializedStyles } from '@emotion/core'
-import { ButtonStyleProps, useCSS } from './Button.style'
-import { BlockIcon } from '../../icons'
+import { ButtonStyleProps, StyledButton, StyledIcon } from './Button.style'
 
 type ButtonProps = {
-  children?: React.ReactNode
-  icon?: boolean
-  disabled?: boolean
+  icon: boolean
+  disabled: boolean
+  hoverable: boolean
   containerCss: SerializedStyles
-  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
-} & ButtonStyleProps
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
+} & Omit<ButtonStyleProps, 'pressedDown' | 'clickable'>
 
 const Button: React.FC<Partial<ButtonProps>> = ({
   children,
   icon,
-  type = 'primary',
+  variant = 'primary',
   disabled = false,
+  hoverable = false,
   containerCss,
-  onClick = () => {},
+  onClick,
   ...styleProps
 }) => {
-  const styles = useCSS({ disabled, type, children, ...styleProps })
+  const [pressedDown, setPressedDown] = useState(false)
```

Comment:
:focus targets elements that are currently on focus, it can't be used to target persistent state like a if a button has been selected. https://developer.mozilla.org/en-US/docs/Web/CSS/:focus

Reply:
Okay so now combined with your other answer I think I see the point. This is to enable sort of functionality of a toggle button, is that correct?

If so, then this functionality shouldn't be enabled for every button and should be instead enabled by props. I'm tempted to say that this should be even a separate component. I don't think a regular button:
a) should have a "pressed down" state
b) should need any state to operate

Reply:
I agree having it on by default is not ideal. 
One solution is to enable the toggling only for animated buttons, in which case `pressedDown` is not updated if `animated` is false. 
The thing I like about having the `pressedDown` as internal state is that this way animated Buttons work like the regular ones and when we consume them we don't have to keep track of the pressedDown state.
What do you think?